cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go

```diff
@@ -192,6 +192,7 @@ func runControlPlanePrepareControlPlaneSubphase(c workflow.RunData) error {
 			data.PatchesDir(),
 			&cfg.ClusterConfiguration,
 			&cfg.LocalAPIEndpoint,
+			false, // isDryRun is false because kubeadm join does not support --dry-run
```

Comment:
i logged https://github.com/kubernetes/kubeadm/issues/2505

```suggestion
			// TODO: add support for join dry-run:
			// https://github.com/kubernetes/kubeadm/issues/2505
			false,
```

Reply:
Done. Thanks!